test_scripts/API/SendLocation/common_send_location.lua

```diff
@@ -0,0 +1,287 @@
+            EXPECT_HMICALL("BasicCommunication.SystemRequest")
+            :Do(function(_, d3)
+                self.hmiConnection:SendResponse(d3.id, "BasicCommunication.SystemRequest", "SUCCESS", { })
+                os.execute("cp /tmp/fs/mp/images/ivsu_cache/PolicyTableUpdate ~/ptu.json")
+                self.hmiConnection:SendNotification("SDL.OnReceivedPolicyUpdate", { policyfile = policy_file_path .. "/" .. policy_file_name })
+              end)
+            mobileSession:ExpectResponse(corIdSystemRequest, { success = true, resultCode = "SUCCESS" })
+          end)
+        :Times(AtMost(1))
+      end
+    end)
+  os.remove(ptu_file_name)
+end
+
+function commonSendLocation.preconditions()
+  commonFunctions:SDLForceStop()
+  commonSteps:DeletePolicyTable()
+  commonSteps:DeleteLogsFiles()
+end
+
+--[[Module functions]]
+
+function commonSendLocation.activate_app(pAppId, self)
+  self, pAppId = commonSendLocation.getSelfAndParams(pAppId, self)
+  if not pAppId then pAppId = 1 end
+  local pHMIAppId = hmiAppIds[config["application" .. pAppId].registerAppInterfaceParams.appID]
+  local mobSession = commonSendLocation.getMobileSession(self, pAppId)
+  local requestId = self.hmiConnection:SendRequest("SDL.ActivateApp", { appID = pHMIAppId })
+  EXPECT_HMIRESPONSE(requestId)
+  mobSession:ExpectNotification("OnHMIStatus", { hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "MAIN" })
+  commonTestCases:DelayedExp(commonSendLocation.minTimeout)
+end
+
+function commonSendLocation.backupHMICapabilities()
+  local hmiCapabilitiesFile = commonFunctions:read_parameter_from_smart_device_link_ini("HMICapabilities")
+  commonPreconditions:BackupFile(hmiCapabilitiesFile)
+end
+
+function commonSendLocation.getDeviceMAC()
+  return config.deviceMAC
+end
+
+function commonSendLocation.getSelfAndParams(...)
+  local out = { }
+  local selfIdx = nil
+  for i,v in pairs({...}) do
+    if type(v) == "table" and v.isTest then
+      table.insert(out, v)
+      selfIdx = i
+      break
+    end
+  end
+  local idx = 2
+  for i = 1, table.maxn({...}) do
+    if i ~= selfIdx then
+      out[idx] = ({...})[i]
+      idx = idx + 1
+    end
+  end
+  return table.unpack(out, 1, table.maxn(out))
+end
+
+function commonSendLocation.getHMIAppId(pAppId)
+  if not pAppId then pAppId = 1 end
+  return hmiAppIds[config["application" .. pAppId].registerAppInterfaceParams.appID]
+end
+
+function commonSendLocation.getMobileSession(self, pAppId)
+  if not pAppId then pAppId = 1 end
+  return self["mobileSession" .. pAppId]
+end
+
+function commonSendLocation.getMobileAppId(pAppId)
+  if not pAppId then pAppId = 1 end
+  return config["application" .. pAppId].registerAppInterfaceParams.appID
+end
+
+function commonSendLocation.getPathToSDL()
+  return config.pathToSDL
+end
+
+function commonSendLocation.postconditions()
+  StopSDL()
+end
+
+function commonSendLocation.rai_ptu(ptu_update_func, self)
```

Comment:
@dboltovskyi , fixed in 6d9eb2c7cc573896229a5a5a5d4b830581e0e427
function is removed - only one fuction will be registerAppWithPTU
Without id it will try to register 1st App.